### ISSUE TITLE
Add more context to `code-block-tools` failures

### ISSUE THREAD

**Issue by stefanvanburen** [rvben/rumdl#521]
### Problem or motivation

I've just enabled code-block-tools with the following config:

```toml
[code-block-tools]
enabled = true

[code-block-tools.languages]
shell = { format = ["shfmt"] }
```

Then, I run:

```console
$ uv run rumdl fmt .
Warning: Code block tool formatting failed: Tool 'shfmt' failed: Exit code 1: <standard input>:3:27: `>` must be followed by a word


Success: No issues found in 313 files (197ms)
```

It seems like at some point `shfmt` failed, but since it's referencing `<standard input>` in the output format, I don't know where the issue is in my files. (I doubt this is specific to `shfmt`.)

### Proposed solution

The warning ought to have more context about which code block it failed at, either printing it out or some better pointer to the file in context. (I tried `--output-format=full` and did not get better results 😄.)

### Feature category

Auto-fix / formatting

### Example markdown

```markdown

```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Fixed in v0.1.49 and v0.1.50. Error messages now include the file path, line number, and language of the failing code block, and use the `file:line:col: [tool] message` format matching rumdl's own lint output. `<standard input>:N:M:` references from tools are translated to absolute file positions. Thanks for reporting, @stefanvanburen!
